sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ContiditionalsShouldStartOnNewLine.cs

```diff
@@ -0,0 +1,70 @@
+/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class ContiditionalsShouldStartOnNewLine : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S3972";
+        private const string MessageFormat = "Move this 'if' to a new line or add the missing 'else'.";
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var ifKeyword = ((IfStatementSyntax)c.Node).IfKeyword;
+
+                var previousSameLineToken = GetPreviousSameLineTokenOrDefault(ifKeyword);
+
+                if (previousSameLineToken != null &&
+                    previousSameLineToken.Value.IsKind(SyntaxKind.CloseBraceToken))
+                {
+                    c.ReportDiagnostic(Diagnostic.Create(rule,
+                            ifKeyword.GetLocation(),
+                            additionalLocations: new[] { previousSameLineToken.Value.GetLocation() }
+                        ));
+                }
+            },
+            SyntaxKind.IfStatement);
+        }
+
+        private static SyntaxToken? GetPreviousSameLineTokenOrDefault(SyntaxToken token)
+        {
+            var previousToken = token.GetPreviousToken();
```

Comment:
But `TryGet...` is a more widely used pattern for such things in the code base. Perhaps I would do it with `TryGet...` despite thinking that I like returning null more...